test/test-parse-statement.cpp

```diff
@@ -578,6 +578,22 @@ TEST(test_parse, else_without_if) {
   }
 }
 
+TEST(test_parse, else_redundant_semicolon) {
+  {
+    spy_visitor v;
+    padded_string code(u8"if (cond) { body; } else; { body; }"_sv);
+    parser p(&code, &v);
+    EXPECT_TRUE(p.parse_and_visit_statement(v));
+    EXPECT_THAT(v.visits, ElementsAre("visit_variable_use",      // cond
+                                      "visit_enter_block_scope",  // (if)
+                                      "visit_variable_use",       // body
+                                      "visit_exit_block_scope")); // (else)
```

Comment:
Nit: The comment is misleading. The `visit_exit_block_scope` is for the `if` body, not the `else` body.
```suggestion
                                      "visit_exit_block_scope")); // (if)
```